pycbc/types/timeseries.py

```diff
@@ -273,8 +273,13 @@ def at_time(self, time, nearest_sample=False, interpolate=None):
 
         if nearest_sample:
             time += self.delta_t / 2.0
-
-        return self[int((time-self.start_time)*self.sample_rate)]
+        deltat = time-self.start_time
+        if hasattr(deltat,'__len__'):
+            if len(deltat) > 0:
+                return self[(deltat*self.sample_rate).astype('int')]
+            else:
+                return _numpy.array([])
+        return self[int(deltat*self.sample_rate)]
```

Comment:
Some more formatting recommendations:
```suggestion
        deltat = time - self.start_time
        if hasattr(deltat, '__len__'):
            if len(deltat) > 0:
                return self[(deltat * self.sample_rate).astype('int')]
            return _numpy.array([])
        return self[int(deltat * self.sample_rate)]
```